src/actions/clusterActions.js

```diff
@@ -300,128 +300,157 @@ export function clusterLoadApps(clusterId) {
       type: types.CLUSTER_INSTALL_APP,
       clusterID,
       app,
     });
 
-    var appsApi = new GiantSwarm.AppsApi();
-    var appConfigsApi = new GiantSwarm.AppConfigsApi();
+    const appsApi = new GiantSwarm.AppsApi();
+    const appConfigsApi = new GiantSwarm.AppConfigsApi();
+    const appSecretsApi = new GiantSwarm.AppSecretsApi();
 
-    var optionalCreateAppConfiguration = new Promise((resolve, reject) => {
-      if (Object.keys(app.valuesYAML).length !== 0) {
-        // If we have user config that we want to create, then
-        // fire off the call to create it.
-        appConfigsApi
+    try {
+      // If we have user config that we want to create, then
+      // fire off the call to create it.
+      Object.keys(app.valuesYAML).length !== 0 &&
+        (await appConfigsApi
           .createClusterAppConfig(clusterID, app.name, {
             body: app.valuesYAML,
           })
-          .then(() => {
-            // The call succeeded, resolve the promise
-            resolve();
-          })
           .catch(error => {
-            if (error.status === 409) {
-              new FlashMessage(
-                `The ConfigMap for <code>${app.name}</code> already exists on cluster <code>${clusterID}</code>`,
-                messageType.ERROR,
-                messageTTL.LONG
-              );
-            } else if (error.status === 400) {
-              new FlashMessage(
-                `Your ConfigMap appears to be invalid. Please make sure all fields are filled in correctly.`,
-                messageType.ERROR,
-                messageTTL.LONG
-              );
-            } else {
-              new FlashMessage(
-                `Something went wrong while trying to create the ConfigMap. Please try again later or contact support: support@giantswarm.io`,
-                messageType.ERROR,
-                messageTTL.LONG
-              );
-            }
-
-            reject(error);
-          });
-      } else {
-        // Otherwise, no user config, so don't do anything.
-        resolve({});
-      }
-    });
-
-    return optionalCreateAppConfiguration
-      .then(() => {
-        return appsApi
-          .createClusterApp(clusterID, app.name, {
-            body: {
-              spec: {
-                catalog: app.catalog,
-                name: app.chartName,
-                namespace: app.namespace,
-                version: app.version,
-              },
-            },
+            appConfigErrorFlashMessages(
+              'ConfigMap',
+              app.name,
+              clusterID,
+              error
+            );
+            throw error;
+          }));
+
+      // If we have an app secret that we want to create, then
+      // fire off the call to create it.
+      Object.keys(app.secretsYAML).length !== 0 &&
+        (await appSecretsApi
+          .createClusterAppSecret(clusterID, app.name, {
+            body: app.secretsYAML,
           })
           .catch(error => {
```

Comment:
I would suggest keeping everything consistent and always using `try/catch` (since we'll need it for stacking multiple promises together in most places)

Reply:
so a nested try catch in this case?

I actually moved this now out to it's own function, still using `.catch` but let me know what you think. Will push in a sec.